Tasks/GoToolV0/gotool.ts

```diff
@@ -119,20 +115,251 @@ function setGoEnvironmentVariables(goRoot: string) {
+    const m = /^(?:go)?(\d+\.\d+\.\d(?:-\d+)?)$/i.exec(ver);
+    if (!m) {
+        throw new Error(`Unexpected version format in Microsoft manifest: ${ver}`);
+    }
+    return m[1];
+}
+
+async function resolveVersionAndCache(version: string, baseUrl?: string): Promise<{ filenameVersion: string, cacheVersion?: string, toolName: string }> {
+    const v = version.trim().replace(/^v/i, '');
+    if (!v) {
+        throw new Error("Version input resolved to empty string.");
+    }
+    
+    // Parse version once for clear classification
+    let parsed: ParsedGoVersion;
+    try {
+        parsed = parseGoVersion(v);
+    } catch (e: any) {
+        throw new Error(`Invalid version format: ${version}`);
+    }
+    
+    const channel = classifyBaseUrl(baseUrl);
+    const isOfficial = channel.type === 'official';
+    const isMicrosoft = channel.type === 'microsoft';
+    // Cache differentiation: use distinct toolName values ('go' vs 'go-aka') so cacheVersion can remain a pure semver.
+    // cacheVersion MUST be a valid semver string accepted by tool-lib; adding prefixes caused null normalization and failures.
+
+    if (isOfficial) {
+        // Official Go: only accepts major.minor or major.minor.patch (no revision)
+        if (parsed.hasRevision()) {
+            throw new Error("Official Go version must be 'major.minor' (e.g. 1.22) or 'major.minor.patch' (e.g. 1.22.3).");
+        }
+        
+        if (parsed.hasPatchOnly()) {
+            // Full patch version specified - use as-is
+            return { filenameVersion: v, cacheVersion: v, toolName: 'go' };
+        }
+        
+        if (parsed.isMajorMinorOnly()) {
+            // Resolve to latest patch
+            let resolved: string | undefined;
+            try {
+                resolved = await getLatestPatchFromGoDev(v);
+            } catch (e: any) {
+                const msg = String(e?.message || e || '');
+                if (/Could not find a stable patch version/i.test(msg)) {
+                    throw new Error(`Requested Go minor ${v} has no stable patch release yet. Specify a full patch once released (e.g. ${v}.0) or use an existing released minor (e.g. 1.23).`);
+                }
+                throw new Error(`Failed to resolve latest patch for ${v}: ${msg}`);
+            }
+            if (!resolved) {
+                throw new Error(`Could not resolve a stable patch for ${v}.`);
+            }
+            return { filenameVersion: resolved, cacheVersion: resolved, toolName: 'go' };
+        }
+        
+        throw new Error("Official Go version must be 'major.minor' (e.g. 1.22) or 'major.minor.patch' (e.g. 1.22.3).");
+    }
+
+    if (isMicrosoft) {
+        // Microsoft aka channel: accepts major.minor, major.minor.patch, or major.minor.patch-revision
+        if (parsed.isMajorMinorOnly()) {
+            // Resolve to latest patch-revision from manifest
+            const resolved = await getMicrosoftLatestFromManifest(v);
```

Comment:
Why pass `v` rather than `parsed`? (Seems related to oddity of `parseGoVersionParts` still existing.)